### PR TITLE
fix(rights) : [CRRE-377] improve rights verification

### PR DESCRIPTION
**Pull request by sjollois**
## Describe your changes
improve rights verification by checking if they have some parameters if they are not admin

## Checklist tests
All functionnalities of CRRE

## Issue ticket number and link
[https://jira.support-ent.fr/browse/CRRE-377](url)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/crre/controllers/OrderRegionController.java [hunk 1]
```diff
@@ -1006,37 +1006,37 @@ private void launchWorker(JsonObject params) {
...
+//                    });
+//                } else {
+//                    ok(request);
+//                }
+//            } else {
+//                log.error(event.left().getValue());
+//                badRequest(request);
+//            }
+//        });
+//    }
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/crre/controllers/OrderRegionController.java:1039]: Pourquoi commenter cet endpoint ? plus utilisé ? cassé ?

File: src/main/java/fr/openent/crre/security/PrescriptorAndStructureRight.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    @Override
+    public void authorize(HttpServerRequest resourceRequest, Binding binding,
+                          UserInfos user, Handler<Boolean> handler) {
+        if(WorkflowActionUtils.hasRight(user, WorkflowActions.ADMINISTRATOR_RIGHT.toString())){
+            handler.handle(true);
+        }else {
+            String idStructure = resourceRequest.getParam("idStructure");
+            handler.handle(WorkflowActionUtils.hasRight(user, WorkflowActions.PRESCRIPTOR_RIGHT.toString()) &&
+                    idStructure != null && !idStructure.equals("null"));
+        }
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/crre/security/PrescriptorAndStructureRight.java:20]: handler.handle(
WorkflowActionUtils.hasRight(user, WorkflowActions.ADMINISTRATOR_RIGHT.toString()) ||
(WorkflowActionUtils.hasRight(user, WorkflowActions.PRESCRIPTOR_RIGHT.toString()) &&
                    resourceRequest.getParam("idStructure") != null && !idStructure.equals("null"))
)

si on souhaite éviter le if / else ;)

File: src/main/java/fr/openent/crre/security/ValidatorAndStructureRight.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    public void authorize(HttpServerRequest resourceRequest, Binding binding,
+                          UserInfos user, Handler<Boolean> handler) {
+        if(WorkflowActionUtils.hasRight(user, WorkflowActions.ADMINISTRATOR_RIGHT.toString())){
+            handler.handle(true);
+        }else {
+            String idStructure = resourceRequest.getParam("idStructure");
+            handler.handle(WorkflowActionUtils.hasRight(user, WorkflowActions.VALIDATOR_RIGHT.toString()) &&
+                    idStructure != null && !idStructure.equals("null"));
+        }
+    }
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/crre/security/ValidatorAndStructureRight.java:21]: cf commentaire sur le [PrescriptorAndStructureRight.java](https://github.com/CGI-OPEN-ENT-NG/crre/pull/9/files#diff-e351b27fefa8884ec80e836d73ff75b7fd0d9b92dc9d2a0008aa8b7206fe1ff8)

File: src/main/java/fr/openent/crre/security/PrescriptorAndStructureRight.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import io.vertx.core.http.HttpServerRequest;
+import org.entcore.common.http.filter.ResourcesProvider;
+import org.entcore.common.user.UserInfos;
+
+public class PrescriptorAndStructureRight implements ResourcesProvider {
+
+    @Override
+    public void authorize(HttpServerRequest resourceRequest, Binding binding,
+                          UserInfos user, Handler<Boolean> handler) {
+        if(WorkflowActionUtils.hasRight(user, WorkflowActions.ADMINISTRATOR_RIGHT.toString())){
```
Comment (c4role) [inline src/main/java/fr/openent/crre/security/PrescriptorAndStructureRight.java:14]: Indentation if / else : il manque les espaces :)

File: src/main/java/fr/openent/crre/security/ValidatorAndStructureRight.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import io.vertx.core.http.HttpServerRequest;
+import org.entcore.common.http.filter.ResourcesProvider;
+import org.entcore.common.user.UserInfos;
+
+public class ValidatorAndStructureRight implements ResourcesProvider {
+
+    @Override
+    public void authorize(HttpServerRequest resourceRequest, Binding binding,
+                          UserInfos user, Handler<Boolean> handler) {
+        if(WorkflowActionUtils.hasRight(user, WorkflowActions.ADMINISTRATOR_RIGHT.toString())){
```
Comment (c4role) [inline src/main/java/fr/openent/crre/security/ValidatorAndStructureRight.java:14]: identation if/else

File: src/main/java/fr/openent/crre/controllers/OrderRegionController.java (hunk 1 above)
Comment (sjollois) [reply src/main/java/fr/openent/crre/controllers/OrderRegionController.java:1039]: plus utilisé oui

Comment (qpic-cgi) [reply src/main/java/fr/openent/crre/controllers/OrderRegionController.java:1039]: Go supprimer alors, non ? (si y'a besoins de retrouver cet endpoint, ce n'est pas un problème car il est dans l'historique du module) ;)

File: src/main/java/fr/openent/crre/controllers/OrderRegionController.java [hunk 6]
```diff
@@ -1006,37 +1006,37 @@ private void launchWorker(JsonObject params) {
...
-                } else {
-                    ok(request);
-                }
-            } else {
-                log.error(event.left().getValue());
-                badRequest(request);
-            }
-        });
-    }
+//    @Get("region/orders/old/status")
```
Comment (Sofianebernoussi) [inline src/main/java/fr/openent/crre/controllers/OrderRegionController.java:1009]: tu veux le laisser commenter ou c'est un oubli?

File: src/main/java/fr/openent/crre/controllers/OrderRegionController.java [hunk 7]
```diff
@@ -1006,37 +1006,37 @@ private void launchWorker(JsonObject params) {
...
+//                }
+//            } else {
+//                log.error(event.left().getValue());
+//                badRequest(request);
+//            }
+//        });
+//    }
 
     int randomStatus() {
         int[] tab = {1, 2, 3, 4, 6, 7, 9, 10, 14, 15, 20, 35, 55, 57, 58, 59};
```
Comment (Sofianebernoussi) [inline src/main/java/fr/openent/crre/controllers/OrderRegionController.java:1042]: tu peux supprimer  cette fonction c'était pour les test de status

File: src/main/java/fr/openent/crre/controllers/StructureController.java [hunk 8]
```diff
@@ -369,10 +366,10 @@ public void updateAmount(final HttpServerRequest request) {
     @Get("/structure/amount")
     @ApiDoc("Get all students amount by structure")
     @SecuredAction(value = "", type = ActionType.RESOURCE)
-    @ResourceFilter(PrescriptorRight.class)
+    @ResourceFilter(PrescriptorAndStructureRight.class)
     public void getAmount(final HttpServerRequest request) {
         try {
-            String id_structure = request.params().get(ID_STRUCTURE);
```
Comment (Sofianebernoussi) [inline src/main/java/fr/openent/crre/controllers/StructureController.java:375]: Tu veu pas utilise la constante ID_STRUCTURE?

Comment (sjollois) [reply src/main/java/fr/openent/crre/controllers/StructureController.java:375]: elle est pas écrit de la même manière

File: src/main/java/fr/openent/crre/controllers/OrderRegionController.java (hunk 6 above)
Comment (sjollois) [reply src/main/java/fr/openent/crre/controllers/OrderRegionController.java:1009]: on l'enlève on s'en sert plus

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**